### PR TITLE
Simplify function list page data fetching

### PR DESCRIPTION
**Pull request by goodoldneon**
## Description

Simplify function list page data fetching. Also fixes duplicate functions in the list during local dev

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/page.tsx [hunk 1]
```diff
@@ -1,88 +1,67 @@
...
-        <FunctionStateFilter handleClick={setFunctionState} selectedOption={functionState} />
-        <FunctionTable rows={functionList.rows} />
+        <FunctionTable rows={rows} />
 
-        {functionList.hasNextPage && (
+        {hasMore !== false && (
           <div className="flex w-full justify-center py-2.5">
             <Button
-              disabled={functionList.isLoading}
+              disabled={isLoading}
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/page.tsx:31]:
```suggestion
              loading={isLoading}
```
Our Buttons have a loading state, which seems better for this situation, as it adds a nice spinner that matches the label you added saying "Loading"

File: ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/useRows.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  const functionsRes = useFunctionsPage({
+    archived,
+    envID: env.id,
+    page,
+  });
+
+  // Append new function data.
+  useEffect(() => {
+    if (functionsRes.data && functionsRes.data.page.page > functionsData.lastPage) {
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/useRows.ts:47]: Does this have to be `functionsRes.data.page.page` ? With the same name it's difficult to understand what each layer returns

Comment (goodoldneon) [reply ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/useRows.ts:47]: That's what our GraphQL endpoint returns. Maybe we can fix that when we standardize pagination?

**Review by anafilipadealmeida — APPROVED** [review]
So much better!

**Pull request merged**